### ISSUE TITLE
Understanding reported circular dependency in Cond*

### ISSUE THREAD

**Issue by davidlange6** [cms-sw/cmssw#31403]
the IBs report a cyclic dep between

![image](https://user-images.githubusercontent.com/5042883/92562907-02c95380-f277-11ea-8d37-07a201e012e2.png)

which I don't see. Eg, 

- CondFormats/MFObjects depends on CondFormats/Serialization and some FWCore stuff. Instead it is used only by CondCore/Utilities and MagneticField/GeomBuilder.

- CondCore/ESSources depends on CondCore/CondDB and some FWCore stuff.

is this loop considering also the plugin of CondFormats/MFObjects or do I miss something?

**Comment by davidlange6** [comment]
assign core

**Comment by davidlange6** [comment]
I see the same sort of effect for 

![image](https://user-images.githubusercontent.com/5042883/92563707-46708d00-f278-11ea-89ee-c2cd8b199103.png)

where 
PhysicsTools/FWLite/src depends on PhysicsTools/SelectorUtils/src 
PhysicsTools/SelectorUtils/bin depends on PhysicsTools/FWLite/src

which to me is ok.

**Comment by smuzaffar** [comment]
yes it (ignominy) looks for all files under a package include plugin and bin directories

**Comment by davidlange6** [comment]
Can we change that behavior? Or is that a feature not a bug?

On Sep 9, 2020, at 8:49 AM, Malik Shahzad Muzaffar <notifications@github.com<mailto:notifications@github.com>> wrote:



yes it (ignominy) looks for all files under a package include plugin and bin directories

**Comment by smuzaffar** [comment]
Currently ignominy treats cmssw package as a unit. Changing this behaviour means we treat every sub-directory (package/plugin, package/bin, package/test etc) of cmssw package as unit. This change is not trivial.

**Comment by davidlange6** [comment]
If it can just ignore test/bin/plugin that would be sufficient I think. Separately one can ensure that no plugin/bin/test is depended on by anything else.

On Sep 9, 2020, at 8:55 AM, Malik Shahzad Muzaffar <notifications@github.com<mailto:notifications@github.com>> wrote:



Currently ignominy treats cmssw package as a unit. Changing this behaviour means we treat every sub-directory (package/plugin, package/bin, package/test etc) of cmssw package as unit. This change is not trivial.

**Comment by smuzaffar** [comment]
yes can be done but as I said, it is not that trivial

**Comment by davidlange6** [comment]
Ok, thanks - I may manage to investigate a simple python script for this (remembering the age of these tools now..)

On Sep 9, 2020, at 9:05 AM, Malik Shahzad Muzaffar <notifications@github.com<mailto:notifications@github.com>> wrote:



yes can be done but as I said, it is not that trivial

**Comment by Dr15Jones** [comment]
A word of warning, we have seen cases of people including from a plugin directory in a different package. We should fix all such cases, but they have happened.
